File: 2019/src/test/java/aoc/Day08Test.java
package aoc;

import org.junit.jupiter.api.*;

import java.io.IOException;
import java.nio.charset.StandardCharsets;
import java.nio.file.Files;
import java.nio.file.Paths;

import static org.junit.jupiter.api.Assertions.assertEquals;

class Day08Test {
    private static String input;
    private Day08 day;

    @BeforeAll
    public static void init() throws IOException {
        input = Files.readString(Paths.get("src/main/resources/day08.txt"), StandardCharsets.UTF_8);
    }

    @BeforeEach
    public void setup() {
        day = new Day08();
    }

    @Nested
    @DisplayName("Part 1")
    class Part1Tests {
        @Test
        void testSolution() {
            long result = day.part1(input);
            assertEquals(2080L, result);
        }
    }

    @Nested
    @DisplayName("Part 2")
    class Part2Tests {
        @Test
        void testSolution() {
            String result = day.part2(input).replaceAll("\r\n", "\n");
            String expected = "" +
                    " ██  █  █ ███   ██  █   █" + "\n" +
                    "█  █ █  █ █  █ █  █ █   █" + "\n" +
                    "█  █ █  █ █  █ █     █ █ " + "\n" +
                    "████ █  █ ███  █      █  " + "\n" +
                    "█  █ █  █ █ █  █  █   █  " + "\n" +
                    "█  █  ██  █  █  ██    █  " + "\n";
            assertEquals(expected, result);
        }
    }
}

File: 2020/aoc2020/src/test/kotlin/Day08Test.kt
import org.junit.jupiter.api.DisplayName
import org.junit.jupiter.api.Nested
import org.junit.jupiter.api.Test
import kotlin.test.assertEquals

@DisplayName("Day 8")
class Day08Test {
    private val example = readText("day08_example.txt")
    private val input = readText("day08.txt")
    private val day = Day08()

    @Nested
    @DisplayName("Part 1")
    inner class Part1 {
        @Test
        fun `Matches example`() {
            val answer = day.part1(example)
            assertEquals(5, answer)
        }

        @Test
        fun `Matches solution`() {
            val answer = day.part1(input)
            assertEquals(1331, answer)
        }
    }

    @Nested
    @DisplayName("Part 2")
    inner class Part2 {
        @Test
        fun `Matches example`() {
            val answer = day.part2(example)
            assertEquals(8, answer)
        }

        @Test
        fun `Matches solution`() {
            val answer = day.part2(input)
            assertEquals(1121, answer)
        }
    }
}
